southbound/impl/src/main/java/org/opendaylight/ansible/southbound/StatusL3vpnListener.java

```diff
@@ -0,0 +1,242 @@
+                       @Nonnull StatusL3vpnProvider statusL3vpn) {
+        LOG.info("remove: id: {}\nstatusL3vpn: {}", instanceIdentifier, statusL3vpn);
+    }
+
+    @Override
+    public void update(@Nonnull InstanceIdentifier<StatusL3vpnProvider> instanceIdentifier,
+                       @Nonnull StatusL3vpnProvider oldStatusL3vpn, @Nonnull StatusL3vpnProvider newStatusL3vpn) {
+        LOG.info("update: id: {}\nold: {}\nnew: {}", instanceIdentifier, oldStatusL3vpn, newStatusL3vpn);
+        // ain't nobody got time to for this
+    }
+
+    private Status createVpn() throws AnsibleCommandException, L3vpnConfigException {
+        Optional<L3vpnSvc> optL3vpnSvc;
+        L3vpnSvc l3vpnSvc;
+        try {
+            optL3vpnSvc = SingleTransactionDataBroker.syncReadOptional(dataBroker, LogicalDatastoreType.CONFIGURATION,
+                    InstanceIdentifier.create(L3vpnSvc.class));
+
+        } catch (ReadFailedException e) {
+            LOG.error("Unable to read L3 VPN config: {}", e.getMessage());
+            return Status.Failed;
+        }
+        if (! optL3vpnSvc.isPresent()) {
+            LOG.error("L3 VPN config is missing");
+            throw new L3vpnConfigException("L3 VPN configuration is missing");
+        }
+        l3vpnSvc = optL3vpnSvc.get();
+        // TODO(trozet): check if sites really belong to Ansible in non-demo implementation
+        // Kick off ansible processes in parallel, then use Futures to callback when it is all done
+
+        List<ListenableFuture<Uuid>> cmdFutures = new ArrayList<>();
+        Sites sites = l3vpnSvc.getSites();
+        if (sites == null) {
+            LOG.error("L3 VPN config missing sites configuration");
+            throw new L3vpnConfigException("L3 VPN config missing sites");
+        }
+        List<Site> siteList = sites.getSite();
+        if (siteList == null || siteList.isEmpty()) {
+            LOG.error("L3 VPN config missing sites configuration");
+            throw new L3vpnConfigException("L3 VPN config missing site configuration");
+        }
+
+        for (Site site: siteList) {
+            if (site == null) {
+                LOG.warn("Null site found in L3 VPN config...ignoring");
+                continue;
+            }
+            LOG.info("Examining site " + site.toString());
+            cmdFutures.add(configureSite(site));
+        }
+
+        l3vpnCommands = Futures.successfulAsList(cmdFutures);
+        l3vpnCommands.addListener(() -> {
+            try {
+                l3vpnCommands.get().forEach(uuid -> {
+                    if (uuid == null) {
+                        txRunner.callWithNewReadWriteTransactionAndSubmit(OPERATIONAL, tx ->
+                            IetfL3vpnSvcUtils.putStatusL3VPN(tx, COMMIT_VERSION, Status.Failed)
+                        );
+                        return;
+                    }
+                });
+                txRunner.callWithNewReadWriteTransactionAndSubmit(OPERATIONAL, tx ->
```

Comment:
Does this really mean they are all done? Couldn't some still be ongoing?

Reply:
I don't think so because I'm using successfulAsList which from what I read returns null if the future didnt complete or errored